### DISCUSSION TITLE
How to show `line numbers` for `code blocks` rendered using the `echo` option?

### DISCUSSION THREAD

**Discussion by hopezh** [observablehq/framework#1256] [Q&A] [answered]
As stated in the title, is there a way, or extension/plugin, to do so? 

An example from Quarto, which renders webpage from markdown files, too: 

![2024-04-17 18 40 02](https://github.com/observablehq/framework/assets/23435810/b5ef6884-b684-41bb-9dc0-1f7302f5064d)

**Comment by Fil** [comment, marked as answer]
We're using [highlight.js](https://highlightjs.org/) whose author is quite [allergic to line numbers](https://highlightjs.readthedocs.io/en/latest/line-numbers.html). There is [a line number plugin](https://github.com/wcoder/highlightjs-line-numbers.js) but plumbing it (or something equivalent) into the call to hljs is not immediate (it happens in [html.ts](https://github.com/observablehq/framework/blob/90b23beb3157b8390a839f693dbfa7c07035b873/src/html.ts#L167)).

Looking at how to use CSS counters to make this work, it appears that the only requirement is to wrap each line into its own DOM element (where currently we're just relying on line feeds). Then a bit of CSS would do the trick.

Feel free to open a feature request (or better yet, a pull request if you manage to make it work!).
